# **Do low-income, predominantly Black communities experience greater food access barriers than other racial groups?**

Supermarket access, economic conditions, and food assistance affect food availability across racial groups, highlighting inequalities in food security. Studying these factors can help address disparities and improve food access in underserved communities. The findings will highlight the disproportionate impact of food insecurity on Black communities and emphasize the need for policy interventions to address racial disparities in food access.




# **Problem Statement:**

Low-income, predominantly Black communities often experience significant barriers to accessing affordable and nutritious food. Many of these neighborhoods are classified as food deserts, where grocery stores are scarce, and fresh food options are limited. As a result, residents frequently rely on convenience stores and fast food, leading to higher rates of food insecurity and diet-related health issues.


# **Data Definition:**

The Food Access Research Atlas provides a geographic analysis of food accessibility in low-income and other census tracts using various measures of supermarket proximity. It includes data on food access for populations within these tracts and offers census-tract-level information that can be used for research or community planning.


https://catalog.data.gov/dataset/food-access-research-atlas


In [1]:
# Import the libraries
import numpy as np                  # Scientific Computing
import pandas as pd                 # Data Analysis
import matplotlib.pyplot as plt     # Plotting
import seaborn as sns               # Statistical Data Visualization

# Let's make sure pandas returns all the rows and columns for the dataframe
pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)

# Force pandas to display full numbers instead of scientific notation
# pd.options.display.float_format = '{:.0f}'.format

# Library to suppress warnings
import warnings
warnings.filterwarnings('ignore')

In [2]:
from google.colab import drive

# Mount Google Drive
drive.mount('/content/drive')

# Define the file path (adjust to your folder location)
file_path = "/content/drive/My Drive/Food_Access_Research_Atlas.csv"

# Read the Excel file
food_access = pd.read_csv(file_path)



Mounted at /content/drive


In [3]:
# Display the first seven rows of the dataframe
food_access.head(7)

CensusTract    State          County  Urban  Pop2010  OHU2010  \
0   1001020100  Alabama  Autauga County      1     1912      693   
1   1001020200  Alabama  Autauga County      1     2170      743   
2   1001020300  Alabama  Autauga County      1     3373     1256   
3   1001020400  Alabama  Autauga County      1     4386     1722   
4   1001020500  Alabama  Autauga County      1    10766     4082   
5   1001020600  Alabama  Autauga County      1     3668     1311   
6   1001020700  Alabama  Autauga County      1     2891     1188   

   GroupQuartersFlag  NUMGQTRS  PCTGQTRS  LILATracts_1And10  \
0                  0       0.0      0.00                  0   
1                  0     181.0      8.34                  1   
2                  0       0.0      0.00                  0   
3                  0       0.0      0.00                  0   
4                  0     181.0      1.68                  0   
5                  0       0.0      0.00                  1   
6                  0      36.0      1.25                  1   

   LILATracts_halfAnd10  LILATracts_1And20  LILATracts_Vehicle  HUNVFlag  \
0                     0                  0                   0         0   
1                     1                  1                   0         0   
2                     0                  0                   0         0   
3                     0                  0                   0         0   
4                     0                  0                   0         1   
5                     1                  1                   0         0   
6                     1                  1                   0         0   

   LowIncomeTracts  PovertyRate  MedianFamilyIncome  LA1and10  LAhalfand10  \
0                0         11.3             81250.0         1            1   
1                1         17.9             49000.0         1            1   
2                0         15.0             62609.0         1            1   
3                0          2.8             70607.0         1            1   
4                0         15.2             96334.0         1            1   
5                1         21.6             69521.0         1            1   
6                1         30.5             39875.0         1            1   

   LA1and20  LATracts_half  LATracts1  LATracts10  LATracts20  \
0         1              1          1           0           0   
1         1              1          1           0           0   
2         1              1          1           0           0   
3         1              1          1           0           0   
4         1              1          1           0           0   
5         1              1          1           0           0   
6         1              1          1           0           0   

   LATractsVehicle_20  LAPOP1_10  LAPOP05_10  LAPOP1_20  LALOWI1_10  \
0                   0     1896.0      1912.0     1896.0       461.0   
1                   0     1261.0      2170.0     1261.0       604.0   
2                   0     1552.0      2857.0     1552.0       478.0   
3                   0     1363.0      3651.0     1363.0       343.0   
4                   1     2643.0      7778.0     2643.0       586.0   
5                   0     3438.0      3668.0     3438.0      1585.0   
6                   0     1231.0      2287.0     1231.0       742.0   

   LALOWI05_10  LALOWI1_20  lapophalf  lapophalfshare  lalowihalf  \
0        467.0       461.0     1912.0          100.00       467.0   
1        962.0       604.0     2170.0          100.00       962.0   
2        971.0       478.0     2857.0           84.70       971.0   
3        893.0       343.0     3651.0           83.24       893.0   
4       1719.0       586.0     7778.0           72.25      1719.0   
5       1674.0      1585.0     3668.0          100.00      1674.0   
6       1307.0       742.0     2287.0           79.12      1307.0   

   lalowihalfshare  lakidshalf  lakidshalfshare  laseniorshalf  \
0         

In [4]:
# Display the bottom seven rows of the dataframe
food_access.tail(7)


CensusTract    State           County  Urban  Pop2010  OHU2010  \
72524  56041975300  Wyoming     Uinta County      0     7761     2696   
72525  56041975400  Wyoming     Uinta County      0     6852     2632   
72526  56043000200  Wyoming  Washakie County      0     3326     1317   
72527  56043000301  Wyoming  Washakie County      1     2665     1154   
72528  56043000302  Wyoming  Washakie County      1     2542     1021   
72529  56045951100  Wyoming    Weston County      0     3314     1322   
72530  56045951300  Wyoming    Weston County      1     3894     1699   

       GroupQuartersFlag  NUMGQTRS  PCTGQTRS  LILATracts_1And10  \
72524                  0     205.0      2.64                  0   
72525                  0      65.0      0.95                  0   
72526                  0      57.0      1.71                  0   
72527                  0      10.0      0.38                  0   
72528                  0      73.0      2.87                  0   
72529                  0     252.0      7.60                  0   
72530                  0      61.0      1.57                  0   

       LILATracts_halfAnd10  LILATracts_1And20  LILATracts_Vehicle  HUNVFlag  \
72524                     0                  0                   0         0   
72525                     0                  0                   1         1   
72526                     0                  0                   0         0   
72527                     0                  0                   0         0   
72528                     0                  0                   0         0   
72529                     0                  0                   0         0   
72530                     0                  0                   0         0   

       LowIncomeTracts  PovertyRate  MedianFamilyIncome  LA1and10  \
72524                0         13.6             62445.0         0   
72525                1         17.3             57248.0         0   
72526                0          9.7             67254.0         1   
72527                0         11.6             64152.0         0   
72528                0         16.3             69605.0         0   
72529                0         17.5             74500.0         1   
72530                0         17.3             76838.0         1   

       LAhalfand10  LA1and20  LATracts_half  LATracts1  LATracts10  \
72524            0         0              0          0           0   
72525            0         0              0          0           0   
72526            1         1              0          0           1   
72527            1         0              1          0           0   
72528            1         0              1          0           0   
72529            1         0              0          0           1   
72530            1         1              1          1           0   

       LATracts20  LATractsVehicle_20  LAPOP1_10  LAPOP05_10  LAPOP1_20  \
72524           0                   0      479.0       479.0        5.0   
72525           0                   1      120.0       120.0        NaN   
72526           1                   1      902.0       902.0      730.0   
72527           0                   0        NaN       743.0        NaN   
72528           0                   0      155.0       892.0      155.0   
72529           0                   0      840.0       840.0      188.0   
72530           0                   0     1376.0      2528.0     1376.0   

       LALOWI1_10  LALOWI05_10  LALOWI1_20  lapophalf  lapophalfshare  \
72524       264.0        264.0         3.0     6037.0           77.78   
72525        24.0         24.0         NaN     5809.0           84.78   
72526       218.0        218.0       176.0     3326.0           99.99   
72527         NaN        279.0         NaN      743.0           27.89   
72528        44.0        261.0        44.0      892.0           35.10   
72529       247.0        247.0        55.0     2709.0           81.74   
72530       384.0 

In [5]:
food_access.shape


(72531, 147)

In [6]:
72531 * 147

10662057

State the shape of the dataframe :

How many rows does the dataframe have?

- 72531 rows

How many columns does the dataframe have?

- 147 columns

What is the total number of datapoints expected in the dataset (rows x columns)?

- 10662057 total datapoints

In [7]:
food_access.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 72531 entries, 0 to 72530
Columns: 147 entries, CensusTract to TractSNAP
dtypes: float64(126), int64(19), object(2)
memory usage: 81.3+ MB


In [8]:
food_access.info(show_counts=True, verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 72531 entries, 0 to 72530
Data columns (total 147 columns):
 #    Column                Non-Null Count  Dtype  
---   ------                --------------  -----  
 0    CensusTract           72531 non-null  int64  
 1    State                 72531 non-null  object 
 2    County                72531 non-null  object 
 3    Urban                 72531 non-null  int64  
 4    Pop2010               72531 non-null  int64  
 5    OHU2010               72531 non-null  int64  
 6    GroupQuartersFlag     72531 non-null  int64  
 7    NUMGQTRS              72506 non-null  float64
 8    PCTGQTRS              72506 non-null  float64
 9    LILATracts_1And10     72531 non-null  int64  
 10   LILATracts_halfAnd10  72531 non-null  int64  
 11   LILATracts_1And20     72531 non-null  int64  
 12   LILATracts_Vehicle    72531 non-null  int64  
 13   HUNVFlag              72531 non-null  int64  
 14   LowIncomeTracts       72531 non-null  int64  
 15   

In [9]:
food_accessColumns = food_access.columns
food_accessColumns

Index(['CensusTract', 'State', 'County', 'Urban', 'Pop2010', 'OHU2010',
       'GroupQuartersFlag', 'NUMGQTRS', 'PCTGQTRS', 'LILATracts_1And10',
       ...
       'TractSeniors', 'TractWhite', 'TractBlack', 'TractAsian', 'TractNHOPI',
       'TractAIAN', 'TractOMultir', 'TractHispanic', 'TractHUNV', 'TractSNAP'],
      dtype='object', length=147)

In [14]:
food_access.isnull().sum()


CensusTract                 0
State                       0
County                      0
Urban                       0
Pop2010                     0
OHU2010                     0
GroupQuartersFlag           0
NUMGQTRS                   25
PCTGQTRS                   25
LILATracts_1And10           0
LILATracts_halfAnd10        0
LILATracts_1And20           0
LILATracts_Vehicle          0
HUNVFlag                    0
LowIncomeTracts             0
PovertyRate                 3
MedianFamilyIncome        748
LA1and10                    0
LAhalfand10                 0
LA1and20                    0
LATracts_half               0
LATracts1                   0
LATracts10                  0
LATracts20                  0
LATractsVehicle_20          0
LAPOP1_10               29957
LAPOP05_10              14540
LAPOP1_20               35914
LALOWI1_10              29957
LALOWI05_10             14540
LALOWI1_20              35914
lapophalf                4568
lapophalfshare           4568
lalowihalf               4568
lalowihalfshare          4568
lakidshalf               4568
lakidshalfshare          4568
laseniorshalf            4568
laseniorshalfshare       4568
lawhitehalf              4568
lawhitehalfshare         4568
lablackhalf              4568
lablackhalfshare         4568
laasianhalf              4568
laasianhalfshare         4568
lanhopihalf              4568
lanhopihalfshare         4568
laaianhalf               4568
laaianhalfshare          4568
laomultirhalf            4568
laomultirhalfshare       4568
lahisphalf               4568
lahisphalfshare          4568
lahunvhalf               4568
lahunvhalfshare          4562
lasnaphalf               4568
lasnaphalfshare          4562
lapop1                  19989
lapop1share             19989
lalowi1                 19989
lalowi1share            19989
lakids1                 19989
lakids1share            19989
laseniors1              19989
laseniors1share         19989
lawhite1                19989
lawhite1share           19989
lablack1                19989
lablack1share           19989
laasian1                19989
laasian1share           19989
lanhopi1                19989
lanhopi1share           19989
laaian1                 19989
laaian1share            19989
laomultir1              19989
laomultir1share         19989
lahisp1                 19989
lahisp1share            19989
lahunv1                 19989
lahunv1share            19966
lasnap1                 19989
lasnap1share            19966
lapop10                 64765
lapop10share            64765
lalowi10                64765
lalowi10share           64765
lakids10                64765
lakids10share           64765
laseniors10             64765
laseniors10share        64765
lawhite10               64765
lawhite10share          64765
lablack10               64765
lablack10share          64765
laasian10               64765
laasian10share          64765
lanhopi10               64765
lanhopi10share          64765
laaian10                64765
laaian10share           64765
laomultir10             64765
laomultir10share        64765
lahisp10                64765
lahisp10share           64765
lahunv10                64765
lahunv10share           64666
lasnap10                64765
lasnap10share           64666
lapop20                 71025
lapop20share            71025
lalowi20                71025
lalowi20share           71025
lakids20                71025
lakids20share           71025
laseniors20             71025
laseniors20share        71025
lawhite20               71025
lawhite20share          71025
lablack20               71025
lablack20share          71025
laasian20               71025
laasian20share          71025
lanhopi20               71025
lanhopi20share          71025
laaian20                71025
laaian20share           71025
laomultir20             71025
laomultir20share        71025
lahisp20                71025
lahisp20share           71025
lahunv20                71025
lahunv20share           70920
lasnap20  

In [13]:
def missing(DataFrame):
    print ('Percentage of missing values in the dataset:\n',
           round((DataFrame.isnull().sum() * 100/ len(DataFrame)),2).sort_values(ascending=False))


# Call the function and execute
# Syntax: missing(DataFrame)
missing(food_access)

Percentage of missing values in the dataset:
 lanhopi20               97.92
lalowi20                97.92
lanhopi20share          97.92
laaian20                97.92
laaian20share           97.92
laomultir20             97.92
laomultir20share        97.92
lahisp20                97.92
lahunv20                97.92
lasnap20                97.92
lapop20                 97.92
lapop20share            97.92
lahisp20share           97.92
lalowi20share           97.92
lawhite20               97.92
lakids20                97.92
laasian20               97.92
lablack20share          97.92
lablack20               97.92
lawhite20share          97.92
laasian20share          97.92
laseniors20             97.92
lakids20share           97.92
laseniors20share        97.92
lahunv20share           97.78
lasnap20share           97.78
laasian10share          89.29
laseniors10             89.29
lablack10share          89.29
lablack10               89.29
lawhite10share          89.29
lawhite10               

**Observations**:

How many missing values are there?

- There are missing values in many of the columns; some variables have missing data of over 97%, while others have moderate missingness of between 27% and 49%.  Some of the columns, such as State, County, and CensusTract, have no missing data, while others have minimal missing values, (<1%).


Are these concentrated in specific rows or columns? How does this affect the analysis?

- The majority of missing values are found in specific columns, particularly those pertaining to race and food access.  Comparing various racial groups may become more difficult as a result, and bias may develop if more data is missing from some communities or regions.

Based on the information that is present, how should the missing values be handled?
How will this affect the analysis?
- Columns missing too much data (over 30%) should be removed. Those with moderate missing data (5%-20%) can be filled in using averages or the most common values. Removing extremely incomplete columns guarantees more trustworthy results.  Although it helps preserve more data, filling in missing values could result in minor mistakes.  The analysis may understate disparities in food access in some communities if missing data is dispersed unevenly.



